### ISSUE TITLE
Win7 - 15khz 120hz 

### ISSUE THREAD

**Issue by psakhis** [flyinghead/flycast#727]
Win7x64 / Vulkan and D3D11 renderers

AMD GPU 360 with modded 15khz drivers on Ryzen 2600

On interlaced resolutions with 15khz, OS represents it with 120hz, on RA log:

[INFO] [Video]: Using HW render, d3d11 driver forced.
[INFO] [Video]: Setting fullscreen to 640x480 @ 120Hz on device \\.\DISPLAY1.

For example on software Display Changer II, i can see it with:
        !--A value of 0/0 represents the optimal refresh rate. Scan-line-order must be Unspecified.--
        refresh numerator="120" denominator="1"/
        !--Unspecified, Progressive, Interlaced (upper field first), Interlaced (lower field first)--
        scanlineorder value="Interlaced (upper field first)"/

Flycast thinking is a 120hz progressive display, and fails with vsync, with 200% slow performance.

This is how switchres switching to @60hz interlaced resolution and not match with flycast is thinking.
[INFO] [CRT]: Requested Resolution: 640x480@60.000000 
[INFO] [CRT]: CRT Mode: 1 - arcade_15 
[INFO] [CRT]: RA Monitor Index Auto: auto
[INFO] [CRT]: SR Disp Monitor Index Auto: Auto  
[INFO] [CRT]: SR rtn 1 
[INFO] [CRT]: Current running core Flycast 
[INFO] Switchres: Modeline "640x480_60i 15.619000KHz 59.957771Hz" 12.980000 640 666 727 831 480 482 488 521 interlace  -hsync -vsync
[INFO] sr_refresh_display: no refresh required
[INFO] set_desktop_mode: \\.\DISPLAY1 (640x480@60) flags(2)
[INFO] sr_switch_to_mode: successfully switched to 640x480@59.957771
[INFO] [CRT]: Setting Aspect Ratio: 1.333333 
[INFO] [CRT]: Setting Video Screen Size to: 640x480 
[INFO] [Video]: Setting refresh rate to: 59.958 Hz.

**Comment by flyinghead** [comment]
Do you have the `Detect Frame Rate Changes` core option enabled?

The Flycast core doesn't request any resolution or frequency change. It reports a 60 Hz (VGA), 59.94 Hz (NTSC) or 50 Hz (PAL) refresh rate to RA.

If the mentioned core option is active, the core may drop to 30 Hz, 15 Hz or less depending on the game but never more than 60 Hz..

**Comment by flyinghead** [comment]
Can you post a standalone log?

**Comment by psakhis** [comment]
What is the command line? i tried --verbose but nothing

**Comment by flyinghead** [comment]
`Settings > Advanced > Other > Log to File`

**Comment by psakhis** [comment]
[flycast_D3D11.log](https://github.com/flyinghead/flycast/files/9478116/flycast_D3D11.log)
[flycast_Vulkan.log](https://github.com/flyinghead/flycast/files/9478117/flycast_Vulkan.log)

In vulkan, works ok if i disable aero desktop composition in Windows, probably because is needed for vsync on this backend.

**Comment by flyinghead** [comment]
With standalone, the screen refresh rate is 60 Hz for both Vulkan and DX11.
```
00:00:012 sdl/sdl.cpp:580 N[RENDERER]: Monitor refresh rate: 60 Hz (640 x 480)
```

**Comment by psakhis** [comment]
Yes, this reports on log. But i see rendering as if it were 120.
Are there any option to disable any 120hz related?

**Comment by flyinghead** [comment]
Vulkan will duplicate frames if the option is enabled (`Settings > Video > Duplicate Frames`) and the monitor refresh rate is 120 Hz or higher. But it wouldn't be active on your machine since the reported refresh rate is 60 Hz.
Again Flycast doesn't change the resolution or refresh rate. Something else on your system must be interfering.

**Comment by psakhis** [comment]
I'm sure, with modded 15khz drivers, windows is reporting x2 values on interlaced resolutions. Problem is here

Maybe i need some hack on windows or src code for bypass it

**Comment by flyinghead** [comment]
The problem might be that the driver VSync runs at 120 Hz? Flycast uses VSync to synchronize with the display (if enabled). If the driver vsyncs at 120 Hz, it will cause sync issues.

**Comment by psakhis** [comment]
Exactly, this is the problem. I don't know if a way to force VSync runs at 60hz

**Comment by psakhis** [comment]
I fixed this issue on W7 editing DWM regkey.

The registry key is HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\DWM\Schedule. The value is FrameRateMin, a DWORD representing the minimum framerate DWM will consider as acceptable. I set it to 120.

**Comment by vkedwardli** [comment]
May be try the new Fixed Video Frequency from dojo?

Details: https://www.patreon.com/posts/fixed-video-68867417
Download: https://github.com/blueminder/flycast-dojo/releases/latest

So you can play with your 15khz setting and disable VSync?

**Comment by psakhis** [comment]
I tried now but with same results without "hacking" DWM. 
At exit, desktop is very laggy, seems dwm don't recognize the real interlaced freq and fails with vsync.
